Gh-364: New getting started guide (#370)
gchq/gaffer-doc: docs/getting-started/example-deployment/project-setup.md

two nodes (vertexes) with one directed edge between them.

```mermaid
graph LR
    A(["Person

        name: marko
        age: 29"])
    --
    "Created
     weight: 0.4"
    -->
    B(["Software

        name: lop
        lang: java"])
```

This data describes one individual and a single piece of software that has been created by that
individual. The data will be loaded into the graph from a CSV file that follows the Neo4j export
syntax, this demonstrates how Gaffer can be used and how it can interact, model and query data from
other popular graph databases. Even with this basic graph we should be able to start building
queries to ask questions like "Who created the software called 'lop'?" and "How much did 'marko'
contribute to the software called 'lop'?" etc.

To go with the diagram above the following CSV file (both raw and rendered are provided) represents
the graph in [Neo4j syntax](https://neo4j.com/labs/apoc/4.4/export/csv/#export-database-csv).

!!! note ""
    Please note that Gaffer often requires additional information about the data, such as
    `:String` on the column headers, to help with typing of the values. This is demonstrated below

>     in the raw file.

Reviewer: For this, it might be useful to link to the "OpenCypher Formats" section of `csv.md` as that's what these extra bits are.

    4. This file holds the schema outlining the elements in the graph, e.g. the
    nodes (aka entities) and edges.
    5. This file defines the different data types in the graph and how they are
    serialised to Java classes.
    6. Config file for additional Gaffer operations and set the class to handle
    them on the store.
    7. The General store properties, sets up what store to use and any additional
    configuration.
    8. This file controls which containers will be started up and the configuration
    of them to ensure correct ports and files are available.

All the files in the `config/accumulo/` and `config/hdfs/` directories will be copied directly from
the two locations in the Gaffer docker repo,
[here](https://github.com/gchq/gaffer-docker/tree/develop/docker/accumulo/conf-2.0.1) and
[here](https://github.com/gchq/gaffer-docker/tree/develop/docker/hdfs/conf). The configuration of
these are out of scope of this example but are covered in other sections of the documentation. The
main focus of this guide will be on the configuration files under the `config/gaffer/` directory.

## Configuration Files

> The Gaffer schema files have a full break down in the [next page](./writing-the-schema.md), this

Reviewer: ```suggestion
There's a full [break down of Gaffer schema files on the next page](./writing-the-schema.md), this
```

section will instead cover the smaller additional config files that go along side the main Gaffer
schema to tweak other aspects of the graph. The location of these files will need to be volume
mounted into the container for them to be included in the deployment which is covered in more detail
[later in the guide](./running-the-deployment.md).

!!! note
    Many of these files have defaults already in the standard `gaffer-rest` container image, but its
    useful to still include them in the project to allow easy configuration.

### Application Properties

This is probably the simplest configuration file in the Gaffer deployment. In general it borrows a
concept from [Spring

> Boot](https://www.tutorialspoint.com/spring_boot/spring_boot_application_properties.htm) to allow

Reviewer: Properties files are not specific to Spring, although the context root change does apply to Spring.

gaffer.store.job.tracker.enabled=true
gaffer.store.operation.declarations=/gaffer/store/operationsDeclarations.json
```

### Operations Declarations

The operation declarations file is a way of enabling additional operations in Gaffer. By default
there are some built in operations already available (the rest API has a get all operations request
to see a list), but its likely you might want to enable others or add your own custom ones. As the
example will load its data from a local CSV file we can activate a couple of additional operations
using the following file.

```json title="operationsDeclarations.json"
{
    "operations": [
        {
            "operation": "uk.gov.gchq.gaffer.operation.impl.export.localfile.ImportFromLocalFile",
            "handler": {
                "class": "uk.gov.gchq.gaffer.store.operation.handler.export.localfile.ImportFromLocalFileHandler"
            }
        },
        {
            "operation": "uk.gov.gchq.gaffer.operation.impl.export.localfile.ExportToLocalFile",
            "handler": {
                "class": "uk.gov.gchq.gaffer.store.operation.handler.export.localfile.ExportToLocalFileHandler"
            }
        }
    ]
}
```

> The two additional operations already exist in Gaffer (you can find them in the code base
> [here](https://github.com/gchq/Gaffer/blob/develop/core/operation/src/main/java/uk/gov/gchq/gaffer/operation/impl/export/localfile/ImportFromLocalFile.java)
> and
> [here](https://github.com/gchq/Gaffer/blob/develop/core/operation/src/main/java/uk/gov/gchq/gaffer/operation/impl/export/localfile/ExportToLocalFile.java)),

Reviewer: These links can be clearer:
```suggestion
The two additional operations already exist in Gaffer (in the code base: 
[ImportFromLocalFile](https://github.com/gchq/Gaffer/blob/develop/core/operation/src/main/java/uk/gov/gchq/gaffer/operation/impl/export/localfile/ImportFromLocalFile.java)
and
[ExportToLocalFile](https://github.com/gchq/Gaffer/blob/develop/core/operation/src/main/java/uk/gov/gchq/gaffer/operation/impl/export/localfile/ExportToLocalFile.java)),
```
